test/functional/pdo_sqlsrv/pdostatement_format_decimals.phpt

```diff
@@ -0,0 +1,385 @@
+        trace("$results[$i], $value\n");
+
+        if ($value !== $results[$i]) {
+            echo "testMoneyTypes: Expected $value but got $results[$i]\n";
+        }
+    }
+}
+
+function testFloatTypes($conn)
+{
+    // This test with the float types of various number of bits, which are retrieved
+    // as numbers by default. When fetched as strings, no formatting is done even with
+    // the statement option FormatDecimals set
+    $epsilon = 0.001;
+    $values = array();
+    for ($i = 0; $i < 5; $i++) {
+        $n1 = rand(1, 100);
+        $n2 = rand(1, 100);
+        $neg = ($i % 2 == 0) ? -1 : 1;
+
+        $n = $neg * $n1 / $n2;
+        array_push($values, $n);
+    }
+
+    $query = "SELECT CONVERT(float(1), $values[0]),
+                     CONVERT(float(12), $values[1]),
+                     CONVERT(float(24), $values[2]),
+                     CONVERT(float(36), $values[3]),
+                     CONVERT(float(53), $values[4])";
+    $stmt = $conn->query($query);
+    $floats = $stmt->fetch(PDO::FETCH_NUM);
+    unset($stmt);
+
+    // Set PDO::SQLSRV_ATTR_FORMAT_DECIMALS to 2 should
+    // have no effect on floating point numbers
+    $numDigits = 2;
+    $options = array(PDO::SQLSRV_ATTR_FORMAT_DECIMALS => $numDigits);
+    $stmt = $conn->prepare($query, $options);
+
+    // By default the floating point numbers are fetched as strings
+    for ($i = 0; $i < 5; $i++) {
+        $stmt->execute();
+        $floatStr = $stmt->fetchColumn($i);
+
+        $floatVal = floatVal($floats[$i]);
+        $floatVal1 = floatval($floatStr);
+        
+        trace("testFloatTypes: $floatVal1, $floatVal\n");
+        
+        // Check if the numbers of decimal digits are the same
+        // It is highly unlikely but not impossible
+        $numbers = explode('.', $floatStr);
+        $len = strlen($numbers[1]);
+        if ($len == $numDigits && $floatVal1 != $floatVal) {
+            echo "Expected $floatVal but $floatVal1 returned. \n";
+        } else {
+            $diff = abs($floatVal1 - $floatVal) / $floatVal;
+            if ($diff > $epsilon) {
+                echo "Expected $floatVal but $floatVal1 returned. \n";
+            }
+        }
+    }
+}
+
+function testMoneyTypes($conn)
+{
+    // With money and smallmoney types, which are essentially decimal types
+    // ODBC driver does not support Always Encrypted feature with money / smallmoney
+    $values = array('24.559', '0', '-0.946', '0.2985', '-99.675', '79.995');
+    $defaults = array('24.5590', '.0000', '-.9460', '.2985', '-99.6750', '79.9950');
+
+    $query = "SELECT CONVERT(smallmoney, $values[0]),
+                     CONVERT(money, $values[1]),
+                     CONVERT(smallmoney, $values[2]),
+                     CONVERT(money, $values[3]),
+                     CONVERT(smallmoney, $values[4]),
+                     CONVERT(money, $values[5])";
+
+    $stmt = $conn->query($query);
+    $results = $stmt->fetch(PDO::FETCH_NUM);

```

Comment:
Fetching the results is done in verifyMoneyValues isn't it? In any case $results is not passed to it...

Reply:
Good point! Let me check